From ff5d9835e5f33a1d39c8167760c7485490e3fc5b Mon Sep 17 00:00:00 2001
From: "Padma, Santhosh Kumar" <skpadma@codeaurora.org>
Date: Wed, 25 Nov 2015 12:30:19 +0530
Subject: [PATCH 31/31] wcnss: Expose dump stack functionality

Add changes to expose dump stack functionality which can be used
by driver to dump stack information when it requires.

CRs-Fixed: 943322
Change-Id: I0fde7142dea2c18daf6b1fb0c5ee4bb8a31a6be0
Signed-off-by: Padma, Santhosh Kumar <skpadma@codeaurora.org>
---
 drivers/net/wireless/wcnss/wcnss_wlan.c | 10 ++++++++++
 include/linux/wcnss_wlan.h              |  2 ++
 2 files changed, 12 insertions(+)

diff --git a/drivers/net/wireless/wcnss/wcnss_wlan.c b/drivers/net/wireless/wcnss/wcnss_wlan.c
index 3bcb71c..1fbbdd6 100644
--- a/drivers/net/wireless/wcnss/wcnss_wlan.c
+++ b/drivers/net/wireless/wcnss/wcnss_wlan.c
@@ -1576,6 +1576,16 @@ int wcnss_get_wlan_unsafe_channel(u16 *unsafe_ch_list, u16 buffer_size,
 }
 EXPORT_SYMBOL(wcnss_get_wlan_unsafe_channel);
 
+/* wlan prop driver cannot invoke show_stack
+ * function directly, so to invoke this function it
+ * call wcnss_dump_stack function
+ */
+void wcnss_dump_stack(struct task_struct *task)
+{
+       show_stack(task, NULL);
+}
+EXPORT_SYMBOL(wcnss_dump_stack);
+
 static int wcnss_smd_tx(void *data, int len)
 {
 	int ret = 0;
diff --git a/include/linux/wcnss_wlan.h b/include/linux/wcnss_wlan.h
index cfa9285..6218c33 100644
--- a/include/linux/wcnss_wlan.h
+++ b/include/linux/wcnss_wlan.h
@@ -15,6 +15,7 @@
 #define _WCNSS_WLAN_H_
 
 #include <linux/device.h>
+#include <linux/sched.h>
 
 enum wcnss_opcode {
 	WCNSS_WLAN_SWITCH_OFF = 0,
@@ -94,6 +95,7 @@ int wcnss_wlan_iris_xo_mode(void);
 void wcnss_flush_work(struct work_struct *work);
 void wcnss_flush_delayed_work(struct delayed_work *dwork);
 int wcnss_get_iris_name(char *iris_version);
+void wcnss_dump_stack(struct task_struct *task);
 
 #ifdef CONFIG_WCNSS_REGISTER_DUMP_ON_BITE
 void wcnss_log_debug_regs_on_bite(void);
-- 
2.5.0

